### PR TITLE
Adds textiles processing.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Fixes cotton not spawning due to having no `chems`.
- Adds raw cotton to the textiles fab intake proc so glittertech spacers can skip the whole process I'm adding below.
- Adds thread, a loom, and a spindle. The spindle takes cotton and produces thread, the loom takes thread and produces cloth.
- Adds `paint_color` to a bunch of cloth prefabs that I forgot to add them to.

## TODO
- [x] Non-placeholder icons for thread.
- [x] Non-placeholder icons for spinning wheel.
- [x] Non-placeholder icons for loom.
- [x] Looping sounds for loom clattering and spindle turning.
- [x] Check if it's actually a spindle or if you meant spinning wheel.

## Future PR stuff
- Replace sutures with a suturer that you load with thread and uses up thread each surgery.
- Add plastic thread to the autolathe for use in suturing.
- Add a needle a la the suturer used to sew complex clothes (instead of in-hand stack crafting).
- Let thread be used to decorate clothing?

## Why and what will this PR improve
Allows low-tech processing of cotton into cloth.
Allows cotton plants to actually do something.

## Authorship
Myself.

## Changelog
:cl:
tweak: Cotton can be harvested and fed to textiles fabricator as a source of cloth.
add: Added loom and spindle for low-tech textiles processing.
/:cl:

### CODE REVIEW

File: code/modules/crafting/textiles/loom.dm [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	if(loaded_thread)
+		loaded_thread.dropInto(loc)
+		loaded_thread = null
+	if(weaving_type)
+		clear_in_progress_weaving()
+	return ..()
+
+/obj/structure/loom/proc/clear_in_progress_weaving()
+	if(weaving_type && weaving_progress > 0)
+		var/obj/item/scrap_material/scraps = new(loc, weaving_type)
```
Comment (out-of-phaze) [inline code/modules/crafting/textiles/loom.dm]: this type doesn't seem to exist? maybe you want `/obj/item/debris/scraps`?

File: code/modules/crafting/textiles/loom.dm [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+	if(!loaded_thread)
+		to_chat(user, SPAN_WARNING("\The [src] needs to be wound with thread before you can weave anything."))
+		return TRUE
+
+	working = TRUE
+	update_icon()
+	var/processed = 0
+	var/last_completion_descriptor = null
+	while(!QDELETED(loaded_thread) && user.do_skilled(2 SECONDS, SKILL_CONSTRUCTION, src))
```
Comment (out-of-phaze) [inline code/modules/crafting/textiles/loom.dm]: understand if this is undesirable atm but maybe make `SKILL_CONSTRUCTION` a var instead, like `/obj/structure/loom/var/skill_used`, so shaded hills can make it use `SKILL_TEXTILES`?

File: code/modules/crafting/textiles/spindle.dm [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		return TRUE
+
+	if(!length(loaded))
+		to_chat(user, SPAN_WARNING("\The [src] needs to be prepared with fibers before you can spin anything."))
+		return TRUE
+
+	working = TRUE
+	update_icon()
+	var/processed = 0
+	while(length(loaded) && user.do_skilled(5 SECONDS, SKILL_CONSTRUCTION, src))
```
Comment (out-of-phaze) [inline code/modules/crafting/textiles/spindle.dm]: see prior comment on the loom about replacing the hardcoded `SKILL_CONSTRUCTION` with a `skill_used` var

File: code/modules/crafting/textiles/spindle.dm [hunk 4]
```diff
@@ -0,0 +1,109 @@
+/obj/structure/spindle
+	name = "spindle"
```
Comment (out-of-phaze) [inline code/modules/crafting/textiles/spindle.dm]: the icon is of a spinning wheel, a spindle is a portable item used to make yarn/thread and it's generally lower-tech than a spinning wheel. might be good to have both options?

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
did some playtesting, cotton thread felt kind of universally worse than grass bundles for a ton more labor, getting cotton on shaded hills didn't really feel like an improvement or tech advancement or anything. not sure if there's something that can be done to adjust that? also cloth bolts just took *so much thread* to make. i get that it's based off of matter but maybe in that case it should be faster, and/or cotton should produce more thread, or maybe we should just make thread/bundles have more matter relative to cloth? (this part isn't actually like, a blocker, just my personal taste/feedback)

File: code/modules/crafting/textiles/loom.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/crafting/textiles/loom.dm]: Branch needs a rebase yeah.

**Comment by MistakeNot4892** [comment]
> did some playtesting, cotton thread felt kind of universally worse than grass bundles for a ton more labor, getting cotton on shaded hills didn't really feel like an improvement or tech advancement or anything. not sure if there's something that can be done to adjust that? also cloth bolts just took _so much thread_ to make. i get that it's based off of matter but maybe in that case it should be faster, and/or cotton should produce more thread, or maybe we should just make thread/bundles have more matter relative to cloth? (this part isn't actually like, a blocker, just my personal taste/feedback)

You can't make clothing or thread from grass, so the 'tech upgrade' is more related to future crafting and material stuff than current utility. Thread being needed for wound suturing is another thing grass won't be much good for.

Per Discord, the processing step for cotton might be somewhere to increase the matter per bundle. Maybe increasing the cloth in chems and making fiber extraction turn reagents into matter? I want to keep everything mediated by matter amounts if possible.

File: code/modules/crafting/textiles/spindle.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/modules/crafting/textiles/spindle.dm]: Yeah I had a todo about this. I might make a drop spindle as a way to do one cotton bundle at a time.

**Comment by PlayerDeer** [comment]
If you remind me tomorrow I can try to do sprites for this.

**Comment by MistakeNot4892** [comment]
https://github.com/NebulaSS13/Nebula/assets/2468979/e72f6b5e-ce37-4e59-8ef0-6a511149d816

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**